### PR TITLE
shot_chart_detail table, simplify GenericRequester api, change default behavior to load current season, SQLite support (kinda), README update

### PR DESCRIPTION
**Pull request by mpope9**
Lots of stuff.

We are fetching the `shot_chart_detail` data in a silly manner. We fetch all games for all players, because that is the path of least resistant, request wise (have to make less total requests). BUT, this means that there will be `game_id`s that _aren't_ in the game table. That breaks the foriegn key constraint. Easiest way around this is building a temp table, inserting all the `shot_chart_detail` into there w/ an index on `game_id` (and not foreign key constraint), then inserting into the real table, filtering on the game ids. I feel like this isn't overkill.

Still finalizing whether we want to use a left join or subquery to insert from the temp table into the `shot_chart_detail`. 🤷 

SQLite _kinda_ works, we just need to fully define the non-unique PKs better, it throws an error without this. Issue here: https://github.com/mpope9/nba-sql/issues/41

ALSO added xactions around the bulk inserting, PeeWee recommends it and I think it makes sense. I don't really see a loss of performance, but I've only been testing with 2020-21.

1996-97 seems broken, not sure if the NBA changed their API, or what. SO, the default behavior is now to load the current season only. Still need to add an `all` option, but I might break that into a second PR because this one is getting big. Issue here: https://github.com/mpope9/nba-sql/issues/37

Also need to add some resiliency and logging around the failed requests. issue for that here: https://github.com/mpope9/nba-sql/issues/38

### CODE REVIEW

File: stats/shot_chart_detail.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import urllib.parse
+
+from models import ShotChartDetail, ShotChartDetailTemp, Game
+from general_requester import GenericRequester
+from peewee import JOIN
+
+
+class ShotChartDetailRequester(GenericRequester):
+
+    shot_char_detail_url = "https://stats.nba.com/stats/shotchartdetail"
```
Comment (avadhanij) [inline stats/shot_chart_detail.py:10]: Typo - `short_chart_detail_url`

File: stats/shot_chart_detail.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        Override method to setup temp table.
+        """
+        super().create_ddl()
+        self.settings.db.bind([ShotChartDetailTemp])
+        self.settings.db.create_tables([ShotChartDetailTemp], safe=True)
+
+    def finalize(self):
+        """
+        This function finishes loading shot_chart_detail by inserting all valid
+        records from the temp table into the main table, then dropping the temp
```
Comment (avadhanij) [inline stats/shot_chart_detail.py:29]: Where is the call to drop the temp table?

**Comment by avadhanij** [comment]
> this means that there will be game_ids that aren't in the game table.

How does this happen? Is it because maybe only a certain season is loaded into the DB?

File: stats/shot_chart_detail.py (hunk 2 above)
Comment (mpope9) [reply stats/shot_chart_detail.py:29]: Good eye, but this uses a table specified with `TEMPORARY` will be dropped at the end of the session. That is defined in the `Meta` section of the Model.

File: stats/shot_chart_detail.py (hunk 1 above)
Comment (mpope9) [reply stats/shot_chart_detail.py:10]: We both typo', will change though.

**Comment by mpope9** [comment]
> > this means that there will be game_ids that aren't in the game table.
> 
> How does this happen? Is it because maybe only a certain season is loaded into the DB?

Yes exactly. We're now only loading the most recent season, and since this fetches all games for a player there is a good chance they don't exist yet.

File: stats/shot_chart_detail.py (hunk 2 above)
Comment (mpope9) [reply stats/shot_chart_detail.py:29]: I'll fix the comment.

File: stats/shot_chart_detail.py (hunk 1 above)
Comment (avadhanij) [reply stats/shot_chart_detail.py:10]: lol

File: stats/shot_chart_detail.py (hunk 2 above)
Comment (avadhanij) [reply stats/shot_chart_detail.py:29]: Ahh, that's very cool.

**Comment by avadhanij** [comment]
So why not fetch only the shot data for the requested seasons?

**Comment by mpope9** [comment]
> So why not fetch only the shot data for the requested seasons? Seems like the API has a season parameter?
> 
> Edit: Ahh I get it, reduce number of requests. Makes sense

Ah yeah acutally I think the endpoint requires both `team_id` and `player_id` to properly function. Just feeding it `season_id` will not return a proper request (i believe, took a few iterations to get this one right.) And a nice side effect is less requests.

**Pull request merged**